aten/src/ATen/autocast_mode.cpp

```diff
@@ -325,6 +325,7 @@ TORCH_LIBRARY_IMPL(aten, Autocast, m) {
   KERNEL(ADD_NS(addmv), "addmv", Tensor (const Tensor &, const Tensor &, const Tensor &, const Scalar&, const Scalar&), lower_precision_fp)
   KERNEL(ADD_NS(addr), "addr", Tensor (const Tensor &, const Tensor &, const Tensor &, const Scalar&, const Scalar&), lower_precision_fp)
   KERNEL(ADD_NS(matmul), "matmul", Tensor (const Tensor &, const Tensor &), lower_precision_fp)
+  KERNEL(ADD_NS(einsum), "einsum", Tensor (c10::string_view, TensorList), lower_precision_fp)
```

Comment:
We'd probably need review from Meta folks on this part. 

The first question on top of my head is that is it legit to apply the same casting rule for all different einsum formulas?

Reply:
Einsum fundamentally does multiplication and addition. The equations merely say which dimensions are multiplied together. The dimensions that are not multiplied are then added together

As both tensor multiplication and summation are both present in the autocast, it seems reasonable that an op that does both operations in a single go also be present.

I am OK in double checking with Meta, though.

Reply:
Added `onnx-needs-import` to get on Meta radar. Is that ok?